src/main/groovy/com/bmuschko/gradle/docker/DockerJavaApplicationPlugin.groovy

```diff
@@ -72,12 +72,15 @@ class DockerJavaApplicationPlugin implements Plugin<Project> {
         project.task(DOCKERFILE_TASK_NAME, type: Dockerfile) {
             description = 'Creates the Docker image for the Java application.'
             dependsOn tarTask
-
             from { dockerJavaApplication.baseImage }
             maintainer { dockerJavaApplication.maintainer }
             addFile({ tarTask.archivePath.name }, { '/' })
-            entryPoint { determineEntryPoint(project, tarTask) }
-            exposePort { dockerJavaApplication.getPorts() }
+            project.afterEvaluate {
+                // pass current delegate on to config closure
+                dockerJavaApplication.exec?.delegate = delegate
```

Comment:
I have some doubts about passing `Dockerfile` as a delegate, because it allows to use _any_ Docker command, i.e.:
````groovy
exec {
   volumes ['/tmp:/tmp']
}
````

Maybe it would be better to provide DSL that offers only `defaultCommand` and `entryPoint`? @cdancy WDYT?